src/olympia/addons/tasks.py

```diff
@@ -31,31 +31,31 @@
 @task
 @use_primary_db
 def version_changed(addon_id, **kw):
-    update_last_updated(addon_id)
-    update_appsupport([addon_id])
-
-
-def update_last_updated(addon_id):
-    queries = Addon._last_updated_queries()
     try:
         addon = Addon.objects.get(pk=addon_id)
     except Addon.DoesNotExist:
         log.info(
             '[1@None] Updating last updated for %s failed, no addon found' % addon_id
         )
         return
+    log.info('[1@None] Updating last updated for %s.' % addon.pk)
+    addon.update(last_updated=compute_last_updated(addon))
+    update_appsupport([addon.pk])
 
-    log.info('[1@None] Updating last updated for %s.' % addon_id)
 
+def compute_last_updated(addon):
```

Comment:
I'm going to have a proper look at this next week but just noting down one thing before I forget: imo `compute_last_updated` should be in utils.py (or elsewhere) as it's not actually a task.

Reply:
Yeah I agree, I'll move it

Reply:
222cee1be4af80fac89b0f4abbb98101290ab0a4